src/font.c

```diff
@@ -547,11 +546,43 @@ void gl_printRaw( const glFont *ft_font,
    i = 0;
    gl_fontRenderStart(stsh, x, y, c);
    while ((ch = u8_nextchar( text, &i )))
-      s = gl_fontRenderGlyph( stsh, ch, c, s );
+      s = gl_fontRenderGlyph( stsh, ch, c, s, forceColor );
    gl_fontRenderEnd();
 }
 
 
+/**
+ * @brief Prints text on screen.
+ *
+ * Defaults ft_font to gl_defFont if NULL.
+ *
+ *    @param ft_font Font to use
+ *    @param x X position to put text at.
+ *    @param y Y position to put text at.
+ *    @param c Colour to use (uses white if NULL)
+ *    @param str String to display.
+ */
+void gl_printRaw( const glFont *ft_font,
+      const double x, const double y,
+      const glColour* c, const char *text )
+{
```

Comment:
Don't think it's a problem, but font rendering is fairly inefficient and slow, and I think this will become much slower. However, don't think it's going to be a real issue (hopefully) on anything except insanely old systems. Might be worth thinking about rewriting this to use shaders.

Reply:
Yes, that would certainly be worth thinking about. Actually @ids1024 mentioned using signed distance fields to do this the proper way.

I'll add a TODO comment to document the shortcoming. It shouldn't be a problem, just something to improve in the future.

🕵️